src/Processors/QtiV2/In/Interactions/ChoiceInteractionMapper.php

```diff
@@ -90,4 +97,47 @@ private function buildOptions(SimpleChoiceCollection $simpleChoices)
         }
         return $options;
     }
+
+    private function buildFeedbackMetadata(SimpleChoiceCollection $simpleChoices)
+    {
+        /* @var $choice SimpleChoice */
+        $metadata = [];
+        foreach ($simpleChoices as $key => $choice) {
+            $flow = $choice->getContent();
+            $class = new \ReflectionClass(get_class($flow));
+            if (property_exists($flow, 'dataPlaceHolder')) {
+                $property = $class->getProperty('dataPlaceHolder');
+                $property->setAccessible(true);
+                $feed = $property->getValue($flow);
+                $count = 0;
+                foreach ($feed as $feeddata) {
```

Comment:
The cyclomatic complexity of this method is quite high - here I would recommend breaking this down into a separate method that you can call inside the `foreach` loop, which takes a `FeedbackInline` object as its argument returns the content to be set to the metadata.

For example:
```php
    ...
    $count = 0;
    foreach ($feed as $feedData) {
        if ($feedData instanceof FeedbackInline) {
            $count++;
            $metadata[] = $this->buildMetadataForFeedbackInline($feedData);
        }
    }
    if ($count == 0) {
        $metadata[] = "";
    }
    ...
```
And the method could do something like this:
```php
protected function buildMetadataForFeedbackInline(FeedbackInline $feedData)
{
    $metadata = "";

    $feedDataArray = array_values((array) $feedData);
    $feedbackArray = array_values((array) $feedDataArray[3]);
    if (sizeof($feedbackArray[0]) >= 2) {
        ...
    } else {
        $feedInlineArray = array_values((array) $feedbackArray[0][0]);
        if (!empty($feedInlineArray[0])) {
            $metadata = trim($feedInlineArray[0]);
        }
    }

    return $metadata;
}
```

Reply:
Done

Reply:
Thanks for suggestion.